### PR TITLE
Add two features to portfolio

### PR DESCRIPTION
**Pull request by bettyashley**
1. Allow user to pick a maximum number of comments to fetch and display.
2. Add button to delete all comments on the page

demo = https://bettyv-step-2020.appspot.com/

### CODE REVIEW

File: portfolio/src/main/java/com/google/sps/servlets/DeleteComment.java [hunk 1]
```diff
@@ -0,0 +1,49 @@
+// Copyright 2019 Google LLC
```
Comment (Enriqe) [inline portfolio/src/main/java/com/google/sps/servlets/DeleteComment.java:1]: same here

File: portfolio/src/main/webapp/script.js [hunk 2]
```diff
@@ -73,14 +73,22 @@ function createListElement(text) {
...
     if(json.length === 0) {
         ulElement.textContent = 'No comments yet.';
     }
 }
+
+/**
+ * Sends a request to the servlet to delete all comments.
+ */
+function deleteComments() {
+    fetch('/deletecomment').then(res => getComments());
```
Comment (Enriqe) [inline portfolio/src/main/webapp/script.js:93]: Do we need to call `getComments()` after deleting them?

File: portfolio/src/main/java/com/google/sps/servlets/ListComment.java [hunk 3]
```diff
@@ -51,18 +53,20 @@ public void doGet(HttpServletRequest request, HttpServletResponse response) thro
...
-        Entity commentEntity = new Entity("Comment");
-        commentEntity.setProperty("comment", comment);
-        commentEntity.setProperty("timestamp", timestamp);
+    private int getCommentCount(HttpServletRequest request){
+        String numCommentString = request.getParameter("num");
 
-        DatastoreService datastore = DatastoreServiceFactory.getDatastoreService();
-        datastore.put(commentEntity);
+        int numComments = 10;
+        try{
```
Comment (Enriqe) [inline portfolio/src/main/java/com/google/sps/servlets/ListComment.java:60]: nit: space after `try`

File: portfolio/src/main/java/com/google/sps/servlets/ListComment.java [hunk 4]
```diff
@@ -51,18 +53,20 @@ public void doGet(HttpServletRequest request, HttpServletResponse response) thro
...
+            numComments = Integer.parseInt(numCommentString);
+        } catch(NumberFormatException e){
```
Comment (Enriqe) [inline portfolio/src/main/java/com/google/sps/servlets/ListComment.java:62]: nit space after `catch` and after closing parenthesis
```suggestion
        } catch (NumberFormatException e) {
```

File: portfolio/src/main/java/com/google/sps/servlets/ListComment.java [hunk 5]
```diff
@@ -51,18 +53,20 @@ public void doGet(HttpServletRequest request, HttpServletResponse response) thro
...
+            System.err.println("This is not a number: " + numCommentString);
+        }
 
-        response.sendRedirect("/");
-  }
+        if(numComments < 0){
```
Comment (Enriqe) [inline portfolio/src/main/java/com/google/sps/servlets/ListComment.java:66]:
```suggestion
        if (numComments < 0) {
```

File: portfolio/src/main/java/com/google/sps/servlets/ListComment.java [hunk 6]
```diff
@@ -51,18 +53,20 @@ public void doGet(HttpServletRequest request, HttpServletResponse response) thro
...
-
-        Entity commentEntity = new Entity("Comment");
-        commentEntity.setProperty("comment", comment);
-        commentEntity.setProperty("timestamp", timestamp);
+    private int getCommentCount(HttpServletRequest request){
+        String numCommentString = request.getParameter("num");
 
-        DatastoreService datastore = DatastoreServiceFactory.getDatastoreService();
-        datastore.put(commentEntity);
+        int numComments = 10;
```
Comment (Enriqe) [inline portfolio/src/main/java/com/google/sps/servlets/ListComment.java:59]: maybe initialize this as `-1`, and if no valid value is provided you will already make it `10` a few lines below

File: portfolio/src/main/java/com/google/sps/servlets/ListComment.java [hunk 7]
```diff
@@ -30,17 +30,19 @@
...
     @Override
     public void doGet(HttpServletRequest request, HttpServletResponse response) throws IOException {
+        int commentCount = getCommentCount(request);
         Query query = new Query("Comment").addSort("timestamp", SortDirection.DESCENDING);
         DatastoreService datastore = DatastoreServiceFactory.getDatastoreService();
         PreparedQuery results = datastore.prepare(query);
 
         List<String> comments = new ArrayList<>();
         for (Entity entity : results.asIterable()) {
+            if (comments.size() == commentCount) break;
```
Comment (Enriqe) [inline portfolio/src/main/java/com/google/sps/servlets/ListComment.java:45]: we can move this check a bit higher and return early so we don't have to even enter this for loop

File: portfolio/src/main/webapp/script.js (hunk 2 above)
Comment (bettyashley) [reply portfolio/src/main/webapp/script.js:93]: It seems like we don't, thanks!

File: portfolio/src/main/java/com/google/sps/servlets/ListComment.java (hunk 7 above)
Comment (Enriqe) [reply portfolio/src/main/java/com/google/sps/servlets/ListComment.java:45]: Discussed offline. I misread the purpose of this line.

**Pull request merged**